File: src/com/cqut/service/module/ModuleService.java
package com.cqut.service.module;

import java.util.List;



import org.apache.ibatis.annotations.Param;

import com.cqut.entity.Module;

public interface ModuleService {

	public int insertModule(Module module);
	
	public int updateModule(Module module);
	
	public int deleteModule(int moduleId);
	
	public List<Module> getModules(String where);
	
	public Module getModuleById(int EId);
	
	public List<Module> searchModuleByName(String name);
}

File: WebContent/cqut/hrms/view/module/js/ModuleService.js
var moduleApp = angular.module('moduleApp', []);

moduleApp.factory('ModuleService',['$resource',function($resource){
	
	var moduleService = $resource('/Module/:action',{});
	
	moduleService.insertModule = function(module,sucesscb,errorcb){
		
		moduleService.save({action:'insertModule'},module,sucesscb,errorcb);
	}
	
	moduleService.updateModule = function(module,sucesscb,errorcb){
		
		moduleService.save({action:'updateModule'},module,sucesscb,errorcb);
	}
	
	moduleService.deleteModule = function(moduleId,sucesscb,errorcb){
		
		moduleService.save({action:'deleteModule'},moduleId,sucesscb,errorcb);
	}
	
	moduleService.getModules = function(sucesscb,errorcb){
		
		moduleService.query({action:'getModules'},sucesscb,errorcb);
	}
	
	moduleService.getModuleById=function(EId,sucesscb,errorcb){
		
		moduleService.save({action:'getModuleById'},EId,sucesscb,errorcb);
	}
	moduleService.searchModuleByName=function(name,sucesscb,errorcb){
		
		moduleService.save({action:'searchModuleByName'},name,sucesscb,errorcb);
	}
	
	return moduleService;
}]);